adminSiteServer/multiDim.ts

```diff
@@ -84,12 +86,27 @@ async function resolveMultiDimDataPageCatalogPathsToIndicatorIds(
         )
     }
 
-    function resolveSingleField(indicator?: IndicatorEntryBeforePreProcessing) {
-        if (typeof indicator === "string") {
-            const indicatorId = catalogPathToIndicatorIdMap.get(indicator)
-            return indicatorId ?? undefined
+    function resolveSingleField(
+        indicator?: IndicatorEntryBeforePreProcessing
+    ): IndicatorConfig | undefined {
+        switch (typeof indicator) {
+            case "number":
+                return { id: indicator }
+            case "string": {
+                const id = catalogPathToIndicatorIdMap.get(indicator)
+                return id ? { id } : undefined
+            }
+            case "object": {
+                if (isIndicatorConfig(indicator)) return indicator
+                if (typeof indicator.id === "string") {
+                    const id = catalogPathToIndicatorIdMap.get(indicator.id)
```

Comment:
I think you'll need to change the definition of `allCatalogPaths` above also, so that `catalogPathToIndicatorIdMap` actually contains the correct mapping for when `indicator` is an object.

Reply:
Nice catch, I was relying too much on type checking. Fixed.